src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -250,6 +250,39 @@ public static bool IsOperationNoneRoot(this IOperation operation)
             }
         }
 
+        /// <summary>
+        /// Gets the first ancestor of this operation with:
+        ///  1. Any OperationKind from the specified <paramref name="ancestorKinds"/>.
+        ///  2. If <paramref name="predicate"/> is non-null, it succeeds for the ancestor.
+        /// Returns null if there is no such ancestor.
+        /// </summary>
+        public static IOperation? GetAncestor(this IOperation root, ImmutableArray<OperationKind> ancestorKinds, Func<IOperation, bool>? predicate = null)
```

Comment:
This is mostly a clone of previous method, except our search looks for any kind from an array of potential ancestor kinds instead of just one.